src/lib/validation/rules.ts

```diff
@@ -0,0 +1,202 @@
+  },
+  password: {
+    validator: validateLoginPassword,
+    errorMessage: '8자 이상 입력해주세요.',
+  },
+};
+
+/**
+ * 로그인 전용 검증 훅
+ */
+export const useLoginValidation = (): ReturnType<typeof useFormValidation> & {
+  isFormValid: (values: Record<string, string>) => boolean;
+} => {
+  const baseValidation = useFormValidation(loginValidationRules);
+
+  // 로그인용 isFormValid (빈 값 허용하지 않음)
+  const isLoginFormValid = useCallback(
+    (values: Record<string, string>): boolean => {
+      return Object.keys(loginValidationRules).every((fieldName) => {
+        const rule = loginValidationRules[fieldName];
+        const value = values[fieldName] || '';
+
+        // 빈 값이면 false 반환
+        if (!value.trim()) {
+          return false;
+        }
+
+        return rule.validator(value);
+      });
+    },
+    []
+  );
+
+  return {
+    ...baseValidation,
+    isFormValid: isLoginFormValid,
+  };
+};
+
+/**
+ * 회원가입 폼 검증 규칙 (안정적인 참조를 위해 함수로 변경)
+ */
+export const getSignupValidationRules = (): ValidationRules => {
+  return {
+    nickname: {
+      validator: validateNickname,
+      errorMessage: '열 자 이하로 작성해주세요.',
+    },
+    email: {
+      validator: validateEmail,
+      errorMessage: '이메일 형식으로 작성해 주세요.',
+    },
+    password: {
+      validator: validatePassword,
+      errorMessage: '8자 이상 입력해주세요.',
+    },
+  };
+};
+
+/**
+ * 비밀번호 확인 검증을 위한 커스텀 훅
+ */
+export const useSignupValidation = (): ReturnType<typeof useFormValidation> & {
+  confirmPasswordError: string;
+  validateConfirmPassword: (
+    password: string,
+    confirmPassword: string
+  ) => boolean;
+  isSignupFormValid: (
+    values: Record<string, string>,
+    options?: { skipConfirmPassword?: boolean }
+  ) => boolean;
+} => {
+  // 검증 규칙을 useMemo로 메모이제이션하여 안정적인 참조 보장
+  const signupRules = useMemo(() => getSignupValidationRules(), []);
+  const baseValidation = useFormValidation(signupRules);
+  const [confirmPasswordError, setConfirmPasswordError] = useState('');
+
+  const validateConfirmPasswordField = useCallback(
+    (password: string, confirmPassword: string): boolean => {
+      // 비밀번호가 비어있으면 검증하지 않음
+      if (!password.trim() || !confirmPassword.trim()) {
+        setConfirmPasswordError('');
+
+        return true;
+      }
+
+      const isValid = password === confirmPassword;
+
+      setConfirmPasswordError(isValid ? '' : '비밀번호가 일치하지 않습니다.');
+
+      return isValid;
+    },
+    []
+  );
+
+  const isSignupFormValid = useCallback(
+    (values: Record<string, string>, { skipConfirmPassword = false } = {}) => {
+      const baseValid = baseValidation.isFormValid(values);
+
+      // skipConfirmPassword가 true이거나 비밀번호가 비어있으면 확인 비밀번호 검증을 건너뛰기
+      const confirmValid =
+        skipConfirmPassword || !values.password || !values.confirmPassword
+          ? true
+          : validateConfirmPasswordField(
+              values.password,
+              values.confirmPassword
+            );
+
+      return baseValid && confirmValid;
```

Comment:
boolean으로 평가되는 값은 'is', 'has'를 접두사로 붙이는 것을 권장드립니다
더불어 복잡한 조건은 변수로 한번 감싸주면 읽기 쉬운 코드가 될 것같아요.

```suggestion
      const isFormValid = baseValidation.isFormValid(values);

      // skipConfirmPassword가 true이거나 비밀번호가 비어있으면 확인 비밀번호 검증을 건너뛰기
      const isPasswordValid = 
      skipConfirmPassword || !values.password || !values.confirmPassword;
      const isConfirmValid = isPasswordValid ? true : validateConfirmPasswordField(
              values.password,
              values.confirmPassword
            );

      return isFormValid && isConfirmValid;
```